### ISSUE TITLE
[suggestion] - town flag

### ISSUE THREAD

**Issue by blek-id** [Zrips/Residence#311]
We do have shop flag, where player can setup shops, have likes and have player able to teleport. Can we have the same feature but a new flag named town? the town can get likes, can set descriptions, and everything similar?

**Comment by ZodoxMC** [comment]
Yes i like the idea, specially since it fits the existing shop "theme" then, maybe there would be more useful for town-use, not quite sure what exactly right now tho
I just opened #312 which is also useful for town-use, if we dont come up with a better idea than this :)

**Comment by Zrips** [comment]
What difference would that be? i mean, what exact use of having residence with town flag would be? Should we make a list of towns in similar way as shops are now? What flags should be enabled/disabled in towns? As much information you can provide, simpler for me to make it reality.

**Comment by blek-id** [comment]
I see a lot of players on my server to want to create towns. Right now there are towns available. But towns aren't as recognized as it is. The access of information to have a town is difficult, unlike shop with is easy to access. By having a town flag, we could mark a residence as a town, which in turn would:
+ make creation of town easier (less admin involvement)
+ recognition of someone's hard work when they create a town is easier to find (via /res town list)
+ residence would have a new functionality with almost similar behavior to shop (probably easy to implement? idk)

Town flag description
Set a town flag by using the normal way of setting flag with /res set. When town flag is true, there's a new behavior for town.

behavior of a town flag:
- Residence is now shown on /res town list command
- /res town list command show a format of something like this:
1. Town name (residence owner name) | Likes/Votes: amount | Members: amount
the description is shown when mouse is hovered on top

new commands for residence:
/res town like -- like the town 
/res town likes -- see number of likes of the town
/res town setdesc -- set description of the town
/res town list -- see list of town available
/res town vote -- vote the town
/res town votes -- see number of votes of the town

New mechanics for town behavior that is not in shop:
Sorting /res list town can be changed to member or likes/vote
Minimum residence size to turn on flag town
Minimum member size to turn on flag town

If there's anything unclear about this, please tell me, english is my second language.

**Comment by Zrips** [comment]
What about flags of that residence? As shop gets tp flag enabled and pvp turned off, to prevent griefing. And then what about actual town requirements? As it would be strange to have town which size is block. So some minimun size should be required, which could be some bonus for player who reached bigger residence limits and now could create town. And then some different mechanics should be applied to that town and players who belong to it. Because otherwise it would be basic residence with label as Town and no difference between them. Maybe requirement to have rentable/on sale areas? 
It is cool idea to make this town thing, but we have to figure out entire mechanic before we can start doing anything with it.

**Comment by blek-id** [comment]
I think that the flag of the residence should be as is? Having pvp turned off is a great idea though. And having town flag requirement would consider the size of the town (probably ignoring Y).

Creating a rentable/on sale area as a requirement is a great idea actually. But again, having the ability to turn some of the requirements off could be useful for some other server owner. I personally never used rent system. Can people rent a place for $0? 

What I think would be unique but simple mechanic for the town flag is:
- having minimum size (ignoring Y)
- having minimum renters (if rent requirement is enabled)
- lease is taken off of bank? To create a mayor with a few members that could gather money together and put it on bank flag.

Enabled flags on default:
bank

Disabled flags on default:
pvp

This is the current idea I have.

**Comment by blek-id** [comment]
Any other feedback?

**Comment by blek-id** [comment]
Sorry for bumping, what do you think?

**Comment by blek-id** [comment]
Bumping yet another time, sorry. Anything you don't like about the proposed idea?

**Comment by blek-id** [comment]
Another bump. Can we have a discussion about this?
